src/block/block_compact.c

```diff
@@ -460,8 +464,21 @@ __wt_block_compact_skip(WT_SESSION_IMPL *session, WT_BLOCK *block, bool *skipp)
     if (WT_VERBOSE_LEVEL_ISSET(session, WT_VERB_COMPACT, WT_VERBOSE_DEBUG_2))
         __block_dump_file_stat(session, block, true);
 
-    __block_compact_skip_internal(
-      session, block, false, block->size, 0, 0, skipp, &block->compact_pct_tenths);
+    /*
+     * Check if the number of available bytes matches the expected configured threshold. Only
+     * perform that check during the first iteration.
+     */
+    if (session->compact->first_pass &&
```

Comment:
We could avoid adding this new first_pass variable and instead check something like `block->compact_pages_reviewed == 0` to indicate the first pass.